crates/pep508-rs/src/marker.rs

```diff
@@ -1345,7 +1345,7 @@ impl MarkerExpression {
                 operator: ExtraOperator::NotEqual,
                 name,
             } => !extras.contains(name),
-            MarkerExpression::Arbitrary { .. } => true,
+            MarkerExpression::Arbitrary { .. } => false,
```

Comment:
If we always evaluate these to false that means that we will skip any requirement that has unknown marker expressions right? It seems like true would _ignore_ the bad marker which makes more sense?

Reply:
"true and warn that it's meaningless" seems like good behavior?

Reply:
I believe this was our behavior until very recently (return `false` here) and is pip's behavior.

Reply:
To be explicit: I think we should omit requirements that have invalid markers, not include them.